hbase-server/src/main/java/org/apache/hadoop/hbase/master/procedure/CloneSnapshotProcedure.java

```diff
@@ -203,6 +216,26 @@ protected Flow executeFromState(final MasterProcedureEnv env, final CloneSnapsho
     return Flow.HAS_MORE_STATE;
   }
 
+  /**
+   * If a StoreFileTracker is specified we strip the TableDescriptor from previous SFT config
+   * and set the specified SFT on the table level
+   */
+  private void updateTableDescriptorWithSFT() {
+    if (StringUtils.isEmpty(customSFT)){
+      return;
+    }
+
+    TableDescriptorBuilder builder = TableDescriptorBuilder.newBuilder(tableDescriptor);
+    builder.setValue(StoreFileTrackerFactory.TRACKER_IMPL, customSFT);
```

Comment:
Do we have any ability to validate that this is a valid `customSFT` prior to modifying the TableDesc? Maybe a call we could make on `StoreFileTrackerFactory`?

Reply:
You are absolutely right, thanks for pointing it out. I'll add a check.

Reply:
I think we should do this check prior to do any real operations? In prepareClone method? In ModifyTableProcedure, we do the check in prepareModify method.

Reply:
I'm moving it prepareClone.